biolink-model.yaml

```diff
@@ -3801,6 +4082,32 @@ classes:
       - "rdf:Statement"
       - "owl:Axiom"
 
+  publication to provider association:
```

Comment:
I think the use case here is to provide a structure to capture more about how a 'Provider' contributes to a publication.  
Consider if we might generalize this idea to define a more generic 'Contribution' object or Association class that can be used to describe how any agent/provider contributes to any type of artifact.  We might look at the Contributor Attribution Model (CAM) here - where a 'Contribution' is akin to the reification of a contributes_to edge that can be used to more richly capture how a contribution to any artifact made (who made it, the specific role they played/type of contribution made,  when they did it, in what organizational context, using what methods / funding, etc). It would cover the things you would put into the qualifier slot defined below, but with more precise semantics.
https://contributor-attribution-model.readthedocs.io/en/latest/

Reply:
Once again @mbrush you are spot on with your assessment. I totally agree. 

My intuition here would be to keep it as an `association` class instead of a `category` class (I think this is what you mean by a generic `Contribution` object here). 

I assume before going on that you've digested revisions committed in https://github.com/biolink/biolink-model/pull/481/commits/e9025fd6d21771a45c20472082cbd97fb9d04ca7 which incorporated your feedback about 'contributor/agent' semantics. This remains slightly relevant here.

I didn't have such extensive use cases in mind when I went this route of creating the `publication to provider association` class (now renamed to `publication to agent association` with `agent` as the `range` of the association) but I agree that some generalization of this association might truly be helpful in a broader Translator context.  

We could reset the `domain` of the association to `named thing` (or leave it unspecified) and rename the association to `entity to agent association`, plus refine the `description` and `slot_usage` accordingly.

Such additional details as organizational context, funding, etc. could be annotated on the edge slots of such an object. BTW, I've been suggesting to Deepak that actually embed the _ad hoc_ TRAPI 'attributes-on-edge-and-slots' annotation protocol into Biolink (see under the **Use of the `attribute` class** comment section in https://github.com/biolink/biolink-model/issues/502).